packages/desktop-client/src/components/reports/Overview.js

```diff
@@ -251,7 +255,65 @@ function CashFlowCard() {
   );
 }
 
+function CategorySpendingCard() {
+  const { getCategories } = useActions();
+
+  const [categories, setCategories] = useState({});
+
+  const end = monthUtils.currentDay();
+  const start = monthUtils.subMonths(end, 3);
+
+  const params = useMemo(() => {
+    return categorySpendingSpreadsheet(
+      start,
+      end,
+      3,
+      (categories.list || []).filter(
+        category => !category.is_income && !category.hidden,
+      ),
+    );
+  }, [start, end, categories]);
+
+  const perCategorySpending = useReport('category_spending', params);
+
+  useEffect(() => {
+    getCategories().then(categories => {
+      setCategories(categories);
+    });
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this will trigger an unnecessary backend call if the categories are already loaded. Would you mind doing it like we did here?

For fetching data: https://github.com/actualbudget/actual/blob/master/packages/desktop-client/src/components/accounts/MobileAccounts.js#L251-L253

Once fetched - for selecting the state slice: https://github.com/actualbudget/actual/blob/master/packages/desktop-client/src/components/accounts/MobileAccounts.js#L236

Bonus points: extract this logic into a reusable hook so we would not need to duplicate the logic.

Reply:
How's that?  And do you want me to update the other places to use the new hook?

Reply:
Ideally: yes, but I won't force you to do this. So up to you :)

Reply:
I've changed the places I found that weren't class-based components now.